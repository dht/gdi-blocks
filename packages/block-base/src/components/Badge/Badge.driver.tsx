import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Badge, BadgeProps } from './Badge';
import { BaseComponentDriver } from 'testing-base';

export class BadgeDriver extends BaseComponentDriver {
    private props: Partial<BadgeProps> = {};

    constructor() {
        super('Badge');
    }

    when: any = {
        rendered: () => {
            render(<Badge {...(this.props as BadgeProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(<Badge {...(this.props as BadgeProps)} />);
        },
    };

    given: any = {
        props: (props: Partial<BadgeProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        WrapperClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
