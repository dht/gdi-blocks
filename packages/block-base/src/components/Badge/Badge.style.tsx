import styled from 'styled-components';

export const Wrapper = styled.div<{ rotation?: number }>`
    flex: 1;

    transform: rotate(${(props) => props.rotation}deg);
`;

export const Image = styled.img`
    width: 170px;
`;
