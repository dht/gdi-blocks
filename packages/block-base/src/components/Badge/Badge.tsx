import React from 'react';
import { Image, Wrapper } from './Badge.style';

export type BadgeProps = {
    rotation?: number;
};

export function Badge(props: BadgeProps) {
    const { rotation = -10 } = props;

    return (
        <Wrapper
            className='Badge-wrapper'
            data-testid='Badge-wrapper'
            rotation={rotation}
        >
            <Image
                src='https://static-b9ebe.web.app/badge-staff-pick.svg'
                alt='Staff Pick'
            />
        </Wrapper>
    );
}

export default Badge;
