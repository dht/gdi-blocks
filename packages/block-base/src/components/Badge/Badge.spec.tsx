import { BadgeDriver } from './Badge.driver';
import Chance from 'chance';

const chance = new Chance();

describe('Badge', () => {
    let driver: BadgeDriver;

    beforeAll(() => {
        driver = new BadgeDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const wrapperClassName = element.get.wrapperClassName();
        const innerText = element.get.label();

        expect(wrapperClassName).toContain('Badge-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('Badge snapshots', () => {
    let driver: BadgeDriver;

    beforeAll(() => {
        driver = new BadgeDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
