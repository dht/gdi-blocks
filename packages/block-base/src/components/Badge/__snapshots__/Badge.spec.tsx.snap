// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshots should match snapshot 1`] = `
<div
  className="sc-bcXHqe geifYo Badge-wrapper"
  data-testid="Badge-wrapper"
>
  <img
    alt="Staff Pick"
    className="sc-gswNZR hPwHSK"
    src="/badge-staff-pick.svg"
  />
</div>
`;
